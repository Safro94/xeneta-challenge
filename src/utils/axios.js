import axios from 'axios';
import https from 'https';

const opts = {
	httpsAgent: new https.Agent({ rejectUnauthorized: false }),
	baseURL: process.env.REACT_APP_SERVER_URL,
};

const interceptorError = error => Promise.reject(error.response);

const instance = axios.create(opts);

instance.interceptors.response.use(response => response, interceptorError);
instance.defaults.headers.common['x-api-key'] = process.env.REACT_APP_API_KEY;

export default instance;
