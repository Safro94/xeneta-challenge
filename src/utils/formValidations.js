const validateSearchForm = ({
	departure,
	destination,
	departureDate,
	returnDate,
}) => {
	if (!departure || !destination || !departureDate) {
		return false;
	}

	if (departure.code === destination.code) {
		return false;
	}

	if (returnDate && returnDate < departureDate) {
		return false;
	}

	return true;
};

export { validateSearchForm };
