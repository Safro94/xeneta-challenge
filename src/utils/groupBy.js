import { marketPositions } from 'constants/marketPositions';

const groupBy = data =>
	data.reduce((accum, item) => {
		if (!accum[marketPositions.low]) accum[marketPositions.low] = [];
		if (!accum[marketPositions.average]) accum[marketPositions.average] = [];
		if (!accum[marketPositions.high]) accum[marketPositions.high] = [];

		accum[marketPositions.low].push({ x: item.day, y: item.low });
		accum[marketPositions.average].push({ x: item.day, y: item.mean });
		accum[marketPositions.high].push({ x: item.day, y: item.high });

		return accum;
	}, {});

export default groupBy;
