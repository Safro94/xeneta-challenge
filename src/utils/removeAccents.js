const removeAccents = value => {
	const map = {
		a: 'á|à|ã|â',
		e: 'é|è|ê',
		i: 'í|ì|î',
		o: 'ó|ò|ô|õ',
		u: 'ú|ù|û|ü',
	};

	for (let pattern in map) {
		value = value.replace(new RegExp(map[pattern], 'g'), pattern);
	}

	return value;
};

export { removeAccents };
