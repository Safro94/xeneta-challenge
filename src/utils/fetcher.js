import axios from 'utils/axios';

const fetcher = async ({ method = 'get', url, data = null }) => {
	return axios[method](url, JSON.parse(data)).then(
		response => response.data,
		error => {
			throw error;
		}
	);
};

export default fetcher;
