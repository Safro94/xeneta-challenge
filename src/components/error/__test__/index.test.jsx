import { render } from 'utils/test-utils';

import Error from '../';

describe('Error', () => {
	it('should match the snapshot', () => {
		const { container } = render(<Error />);

		expect(container.firstElementChild).toMatchSnapshot();
	});
});
