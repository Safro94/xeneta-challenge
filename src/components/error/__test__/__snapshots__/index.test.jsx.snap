// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match the snapshot 1`] = `
<div
  class="container"
  role="alert"
>
  <img
    alt="Error"
    class="image"
    src="error.png"
  />
  <h2>
    Ooops, there's been an error, please try again in a few minutes
  </h2>
</div>
`;
